### PR TITLE
task was solved

### PR DESCRIPTION
**Pull request by vanyachyzh**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,36 @@
...
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const item in actions[i].extraData) {
+        copy[item] = actions[i].extraData[item];
+      }
+
+      resultArray.push({
+        ...copy,
+      });
```
Comment (IvanFesenko) [inline src/transformStateWithClones.js:21]: Do not repeat your self, do this action once at the end

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Good job. Please fix code duplicating

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copy = { ...state };
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
```
Comment (AlisaVlasova) [inline src/transformStateWithClones.js:14]: use switch/case

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,28 @@
...
+      for (const item in actions[i].extraData) {
```
Comment (AlisaVlasova) [inline src/transformStateWithClones.js:15]: you can use Object.assign instead for loop

**Review by AlisaVlasova — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,32 @@
...
+      case actions[i].type === 'addProperties':
+        Object.assign(copy, actions[i].extraData);
+        break;
+
+      case actions[i].type === 'removeProperties':
+        for (let j = 0; j < actions[i].keysToRemove.length; j++) {
+          delete copy[actions[i].keysToRemove[j]];
+        };
+
+        break;
```
Comment (OlEzhA0) [inline src/transformStateWithClones.js:24]:
```suggestion
    switch (actions[i].type) {
      case 'addProperties':
        Object.assign(copy, actions[i].extraData);
        break;

      case 'removeProperties':
        for (let j = 0; j < actions[i].keysToRemove.length; j++) {
          delete copy[actions[i].keysToRemove[j]];
        };

        break;
```

File: src/transformStateWithClones.js [hunk 5]
```diff
@@ -7,7 +7,32 @@
...
+
+      default:
+        copy = {};
+    }
+
+    resultArray.push({
+      ...copy,
+    });
```
Comment (OlEzhA0) [inline src/transformStateWithClones.js:32]: ?
```suggestion
    resultArray.push(copy);
```

**Review by OlEzhA0 — CHANGES_REQUESTED** [review]

**Review by Viktor-Kost — APPROVED** [review]

**Pull request closed without merging**